src/lazy/binary/raw/value.rs

```diff
@@ -179,7 +185,7 @@ impl<'data> LazyRawBinaryValue<'data> {
 
     /// If this value is within a struct, returns its associated field name as a `Some(SymbolID)`.
     /// Otherwise, returns `None`.
-    pub(crate) fn field_name(&self) -> Option<SymbolId> {
+    pub(crate) fn field_id(&self) -> Option<SymbolId> {
```

Comment:
🗺️ I renamed this binary reader method from `field_name` to `field_id` to avoid be confused with the raw reader trait of the same name.

The binary reader can always return a symbol ID representing the field name if there is one. However, the raw reader trait needs to accommodate failure cases in the raw text reader and returns an `IonResult<RawSymbolTokenRef>` instead of an `Option<SymbolId>`.